src/control/server/server_utils.go

```diff
@@ -500,9 +500,18 @@ func registerLeaderSubscriptions(srv *server) {
 				srv.log.Debugf("%s marked rank %d:%x dead @ %s", evt.Hostname, evt.Rank, evt.Incarnation, ts)
 				// Mark the rank as unavailable for membership in
 				// new pools, etc. Do group update on success.
-				if err := srv.membership.MarkRankDead(system.Rank(evt.Rank), evt.Incarnation); err == nil {
-					srv.mgmtSvc.reqGroupUpdate(ctx, false)
+				if err := srv.membership.MarkRankDead(system.Rank(evt.Rank), evt.Incarnation); err != nil {
+					srv.log.Errorf("failed to mark rank %d:%x dead: %s", evt.Rank, evt.Incarnation, err)
+					if system.IsNotLeader(err) {
+						// If we've lost leadership while processing the event,
+						// attempt to re-forward it to the new leader.
+						evt = evt.WithForwarded(false).WithForwardable(true)
```

Comment:
Just for my understanding, why WithForwaded is set to false ?
Naively, I was expecting the opposite.

Reply:
We want it to be forwarded so in this line we are saying it hasn't already been forwarded (which will mean it does get forwarded). If forwarded == true, event will not be forwarded again.